### PR TITLE
configure: format configure --help text

### LINKED ISSUE #6328: configure: help message not formated correctly
For example:

```
AC_ARG_WITH(pmilib,
     AS_HELP_STRING([--with-pmilib=name],
                [Specify the PMI library. Accepted options are:
                   mpich - use MPICH PMI library (default).
                   slurm - use Slurm PMI library. Use --with-slurm=path to configure
                           Slurm library path.
                   cray  - use Cray PMI library. Use --with-craypmi=path to configure
                           Cray PMI library path.
                   pmix  - use PMIx library. Use --with-pmix=path to configure
                           PMIx library path.
                ]),,
```
 
But the output from “./configure –help” is:

 
```
  --with-pmilib=name      Specify the PMI library. Accepted options are: mpich
                                         - use MPICH PMI library (default). slurm - use Slurm
                                         PMI library. Use --with-slurm=path to configure
                                         Slurm library path. cray - use Cray PMI library. Use
                                         --with-craypmi=path to configure Cray PMI library
                                         path. pmix - use PMIx library. Use --with-pmix=path
                                         to configure PMIx library path.
 ```

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description
Some of the help text are formated incorrectly.

Fixes #6328 
[skip warnings]
## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by hzhou** [comment]
Resulting output form `./configure --help`:
[config_help.txt](https://github.com/pmodels/mpich/files/11031092/config_help.txt)

File: configure.ac [hunk 1]
```diff
@@ -1448,16 +1448,18 @@ fi
...
-	with_pmilib=mpich)
+AC_ARG_WITH(pmi, [
+  --with-pmi=name - Specify the pmi interface for MPICH.
+                      pmi1 - use PMIv1 (default)
+                      pmi2 - use PMI2
+                      pmix - use PMIx
+],, with_pmi=pmi1)
+
+AC_ARG_WITH(pmilib, [
+  --with-pmilib=option -  [Deprecated] Specify the PMI library. Use --with-pmi=[PATH],
```
Comment (raffenet) [inline configure.ac:1459]: What is the preferred way to specify the path to an external PMI library? When setting up Polaris CI, I still ended up using this option to point to link with the Cray PMI library.

**Review by raffenet — APPROVED** [review]
This looks good. My one question is more about usage.

File: configure.ac (hunk 1 above)
Comment (hzhou) [reply configure.ac:1459]: Does `--with-pmi2=[cray-lib-path]` work?

Comment (raffenet) [reply configure.ac:1459]: > Does `--with-pmi2=[cray-inc-lib-path]` work?

It does after I rebased onto the latest `main`. I got bitten using an older branch.

**Pull request merged**